.github/workflows/e2e-macos.yml

```
@@ -0,0 +1,80 @@
+# SPDX-License-Identifier: Apache-2.0
+
+name: E2E test on macOS ARM64
+
+on:
+    workflow_dispatch:
+    push:
+        branches:
+            - "main"
+            - "release-**"
+        paths:
+            - "**.py"
+            - "pyproject.toml"
+            - "requirements*.txt"
+            - "tox.ini"
+            - "scripts/*.sh"
+            - ".github/**"
+    pull_request:
+        branches:
+            - "main"
+            - "release-**"
+        paths:
+            - "**.py"
+            - "pyproject.toml"
+            - "requirements*.txt"
+            - "tox.ini"
+            - "scripts/*.sh"
+            - ".github/**"
+
+jobs:
+    e2e:
+        name: "E2E test on ${{ matrix.platform }} (Python ${{ matrix.python }})"
+        runs-on: "${{ matrix.platform }}"
+        strategy:
+            matrix:
+                python:
+                    - "3.11"
+                platform:
+                    - "macos-latest"
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v4
+              with:
+                  # https://github.com/actions/checkout/issues/249
+                  fetch-depth: 0
+                  submodules: "true"
+
+            - name: Install Packages
+              run: |
+                  brew install expect coreutils bash
+
+            - name: Setup Python ${{ matrix.python }}
+              uses: actions/setup-python@v5
+              with:
+                  python-version: ${{ matrix.python }}
+                  cache: pip
+                  cache-dependency-path: |
+                      **/pyproject.toml
+                      **/requirements*.txt
+
+            - name: Cache huggingface
+              uses: actions/cache@v4
+              with:
+                  path: ~/.cache/huggingface
+                  # config contains DEFAULT_MODEL
+                  key: huggingface-${{ hashFiles('src/instructlab/config.py') }}
+
+            - name: Install ilab
+              run: |
+                  python3 -m venv venv
+                  . venv/bin/activate
+                  sed 's/\[.*\]//' requirements.txt > constraints.txt
+                  python3 -m pip cache remove llama_cpp_python
+                  CMAKE_ARGS="-DLLAMA_METAL=on" python3 -m pip install --no-binary llama_cpp_python -c constraints.txt llama_cpp_python
```

```suggestion
                  CMAKE_ARGS="-DLLAMA_METAL=off" python3 -m pip install --no-binary llama_cpp_python -c constraints.txt llama_cpp_python
```

Did you mean to disable it instead since the runner does not support it? I've seen the error and this is how I worked around it https://github.com/instructlab/instructlab/blob/main/scripts/functional-tests.sh#L60-L71.

---

I was hoping to get CI working with hardware acceleration.

Let's give your suggestion a try!